File: app_java/src/main/java/com/dwp/mvvmbasicsample/TaskRepository.java
package com.dwp.mvvmbasicsample;

import android.app.Application;
import android.os.AsyncTask;

import androidx.lifecycle.LiveData;

import com.dwp.mvvmbasicsample.room.Task;
import com.dwp.mvvmbasicsample.room.TaskDao;
import com.dwp.mvvmbasicsample.room.TaskDatabase;

import java.util.List;

/**
 * Created by dwp on 2020-11-03.
 */
public class TaskRepository {
    private TaskDao taskDao;
    private LiveData<List<Task>> allTasks;

    public TaskRepository(Application application) {
        TaskDatabase database = TaskDatabase.getInstance(application);
        taskDao = database.taskDao();
        allTasks = taskDao.getAllTasks();
    }

    public void insert(Task task) {
        new InsertTaskAsyncTask(taskDao).execute(task);
    }

    public void update(Task task) {
        new UpdateTaskAsyncTask(taskDao).execute(task);
    }

    public void delete(Task task) {
        new DeleteTaskAsyncTask(taskDao).execute(task);
    }

    public void deleteAllTasks() {
        new DeleteAllTasksAsyncTask(taskDao).execute();
    }

    public LiveData<List<Task>> getAllTasks() {
        return allTasks;
    }

    private static class InsertTaskAsyncTask extends AsyncTask<Task, Void, Void> {

        private TaskDao taskDao;

        public InsertTaskAsyncTask(TaskDao taskDao) {
            this.taskDao = taskDao;
        }

        @Override
        protected Void doInBackground(Task... tasks) {
            taskDao.insert(tasks[0]);
            return null;
        }
    }

    private static class UpdateTaskAsyncTask extends AsyncTask<Task, Void, Void> {

        private TaskDao taskDao;

        public UpdateTaskAsyncTask(TaskDao taskDao) {
            this.taskDao = taskDao;
        }

        @Override
        protected Void doInBackground(Task... tasks) {
            taskDao.update(tasks[0]);
            return null;
        }
    }

    private static class DeleteTaskAsyncTask extends AsyncTask<Task, Void, Void> {

        private TaskDao taskDao;

        public DeleteTaskAsyncTask(TaskDao taskDao) {
            this.taskDao = taskDao;
        }

        @Override
        protected Void doInBackground(Task... tasks) {
            taskDao.delete(tasks[0]);
            return null;
        }
    }

    private static class DeleteAllTasksAsyncTask extends AsyncTask<Void, Void, Void> {

        private TaskDao taskDao;

        public DeleteAllTasksAsyncTask(TaskDao taskDao) {
            this.taskDao = taskDao;
        }

        @Override
        protected Void doInBackground(Void... voids) {
            taskDao.deleteAllTasks();
            return null;
        }
    }
}

File: app_kotlin/src/main/java/com/dwp/mvvmbasicsample/TaskRepository.kt
package com.dwp.mvvmbasicsample

import android.app.Application
import android.os.AsyncTask
import androidx.lifecycle.LiveData
import com.dwp.mvvmbasicsample.room.Task
import com.dwp.mvvmbasicsample.room.TaskDao
import com.dwp.mvvmbasicsample.room.TaskDatabase

/**
 * Created by dwp on 2020-10-30.
 */
class TaskRepository(application: Application) {
    private val database = TaskDatabase.getInstance(application)
    private val taskDao = database.taskDao()
    private val allTasks = taskDao.getAllTasks()

    fun insert(task: Task) {
        InsertTaskAsyncTask(taskDao).execute(task)
    }

    fun update(task: Task) {
        UpdateTaskAsyncTask(taskDao).execute(task)
    }

    fun delete(task: Task) {
        DeleteTaskAsyncTask(taskDao).execute(task)
    }

    fun deleteAllTasks() {
        DeleteAllTasksAsyncTask(taskDao).execute()
    }

    fun getAllTasks(): LiveData<MutableList<Task>> {
        return allTasks
    }

    private class InsertTaskAsyncTask(val taskDao: TaskDao) : AsyncTask<Task, Void, Void>() {
        override fun doInBackground(vararg tasks: Task): Void? {
            taskDao.insert(tasks[0])
            return null
        }
    }

    private class UpdateTaskAsyncTask(val taskDao: TaskDao) : AsyncTask<Task, Void, Void>() {
        override fun doInBackground(vararg tasks: Task): Void? {
            taskDao.update(tasks[0])
            return null
        }
    }

    private class DeleteTaskAsyncTask(val taskDao: TaskDao) : AsyncTask<Task, Void, Void>() {
        override fun doInBackground(vararg tasks: Task): Void? {
            taskDao.delete(tasks[0])
            return null
        }
    }

    private class DeleteAllTasksAsyncTask(val taskDao: TaskDao) : AsyncTask<Void, Void, Void>() {
        override fun doInBackground(vararg voids: Void): Void? {
            taskDao.deleteAllTasks()
            return null
        }
    }
}